bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/handler/SimpleDeviceHandler.java

```diff
@@ -0,0 +1,251 @@
+     */
+    protected void assertChannel(SolarwattChannel solarwattChannel) {
+        ChannelUID channelUID = new ChannelUID(this.getThing().getUID(), solarwattChannel.getChannelName());
+        ChannelTypeUID channelType = this.channelTypeProvider.assertChannelType(solarwattChannel);
+        if (this.getThing().getChannel(channelUID) == null) {
+            ThingBuilder thingBuilder = this.editThing();
+            thingBuilder.withChannel(getChannelBuilder(solarwattChannel, channelUID, channelType).build());
+
+            this.updateThing(thingBuilder.build());
+        }
+    }
+
+    /**
+     * Get a builder for a channel type according to the {@link SolarwattChannel}
+     *
+     * @param solarwattChannel channel type definition
+     * @param channelUID uid of the channel
+     * @param channelType uid of the channel type
+     * @return builder for that channel type
+     */
+    public static ChannelBuilder getChannelBuilder(SolarwattChannel solarwattChannel, ChannelUID channelUID,
+            ChannelTypeUID channelType) {
+        String itemType = CoreItemFactory.STRING;
+        Unit<?> unit = solarwattChannel.getUnit();
+        if (unit != null) {
+            String dimension = UnitUtils.getDimensionName(unit);
+
+            if (Units.PERCENT.equals(unit)) {
+                // strangely it is Angle
+                dimension = ":Dimensionless";
+            }
+
+            itemType = CoreItemFactory.NUMBER;
+            if (dimension != null && !dimension.isEmpty()) {
+                itemType = CoreItemFactory.NUMBER + ":" + dimension;
+            }
+        }
+        ChannelBuilder channelBuilder = ChannelBuilder.create(channelUID, itemType);
+
+        channelBuilder.withLabel(solarwattChannel.getChannelName()).withType(channelType).withDescription(MessageFormat
+                .format("Value for {0} with Unit: {1}", solarwattChannel.getChannelName(), solarwattChannel.getUnit()))
+                .withKind(ChannelKind.STATE);
+        return channelBuilder;
+    }
+
+    /**
+     * Get the {@link EnergyManagerHandler}.
+     *
+     * Only the {@link EnergyManagerHandler} has knowledge about the devices itself.
+     *
+     * @return instance responsible for this handler
+     */
+    protected @Nullable EnergyManagerHandler getEnergyManagerHandler() {
+        Bridge bridge = this.getBridge();
+        if (bridge != null) {
+            BridgeHandler bridgeHandler = bridge.getHandler();
+            if (bridgeHandler instanceof EnergyManagerHandler) {
+                return (EnergyManagerHandler) bridgeHandler;
+            } else {
+                // happens while dynamically reloading the binding
+                this.logger.error("BridgeHandler is not implementing EnergyManagerHandler {}", bridgeHandler);
+            }
+        } else {
+            // this handler can't work without a bridge
+            this.logger.error("Bridge is null");
```

Comment:
If there is a configuration error, you should change the thingStatus and not use the logger here.

Reply:
Moved to chaging thingStatus